artcommon/artcommonlib/redis.py

```diff
@@ -65,19 +88,40 @@ async def set_value(conn: aioredis.commands.Redis, key: str, value):
     await conn.set(key, value)
 
 
+@handle_connection_sync
+def set_value_sync(conn: redis.client.Redis, key: str, value):
+    """
+    Same as set_value(), but synchronous
+    """
+
+    logger.debug('Setting key %s to %s', key, value)
+    conn.set(key, value)
+
+
 @handle_connection
-async def get_keys(conn: aioredis.commands.Redis, pattern: str):
+async def get_keys(conn: redis.asyncio.client.Redis, pattern: str):
     """
     Returns a list of keys (string) matching pattern (e.g. "*.count")
     """
 
-    keys = await conn.keys(pattern)
+    keys = [key.decode() for key in await conn.keys(pattern)]
+    logger.debug('Found keys matching pattern %s: %s', pattern, ', '.join(keys))
+    return keys
+
+
+@handle_connection_sync
+def get_keys_sync(conn: redis.client.Redis, pattern: str):
+    """
+    Same as get_keys(), but synchronous
+    """
+
+    keys = [key.decode() for key in conn.keys(pattern)]
```

Comment:
Can remove the `decode()`
And if a key is not found, I think None will be returned? So maybe we return a dict?

Reply:
Or list is fine as well, I was thinking of set first and then thought of dict. No strong opinion

Reply:
yeah I took that out after setting `decode_responses=True`